Quick start guide for next.js SDK (#260)
imagekitio/docs-keep-a-changelog: getting-started/quickstart-guides/nextjs.md

---
description: >-
  Real-time image & video resizing, automatic optimization, and file uploading in Next.js using ImageKit.io.
---

# Next.js

This quick start guide demonstrates how to integrate ImageKit into a Next.js application. The code samples provided are available on GitHub: [https://github.com/imagekit-samples/quickstart/tree/master/next](https://github.com/imagekit-samples/quickstart/tree/master/next).

This guide covers the following topics:

- [Setting up ImageKit Next.js SDK](nextjs.md#setup-imagekit-next.js-sdk)
- [Rendering images](nextjs.md#rendering-images)
- [Setting the ImageKit context for the SDK](nextjs.md#setting-imagekit-context-for-the-sdk)
- [Applying common image manipulations](nextjs.md#basic-image-manipulation)
- [Adding overlays](nextjs.md#adding-overlays)
- [Lazy loading images](nextjs.md#lazy-loading-images-in-next.js)
- [Blurred image placeholder](nextjs.md#blurred-image-placeholder)
- [Client-side file uploading](nextjs.md#uploading-files-in-next.js)
- [Advanced file uploading](nextjs.md#advanced-file-upload)
- [Rendering videos](nextjs.md#rendering-videos)

## **Setup ImageKit Next.js SDK**

> For this tutorial, it is recommended to create a dummy Next.js app, as demonstrated below.

Reviewer: ```suggestion
For this tutorial, let's create a dummy Next.js app, as demonstrated below.
```

**Create a Next.js app:**

Let's use the `create-next-app` CLI utility provided by Next.js to build a new project:

```bash
npx create-next-app@latest imagekit-next-app
```

We will be using the below configuration for the dummy app.

```js
✔ Would you like to use TypeScript?  No
✔ Would you like to use ESLint? Yes
✔ Would you like to use Tailwind CSS? No
✔ Would you like to use `src/` directory? No
✔ Would you like to use App Router? (recommended) Yes
✔ Would you like to customize the default import alias (@/*)? No
```

Navigate to the project directory:

```
cd imagekit-next-app/
```

> Open up the project in your text editor of choice, and navigate to `app/page.js`. This is where we will do most of our work. It should look like this:

Reviewer: ```suggestion
Open up the project in your preferred text editor, and navigate to app/page.js. This is where we will focus our efforts. The file should look like this:
```
